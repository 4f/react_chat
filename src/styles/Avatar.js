export default theme => ({
  avatar: {
    "&:hover": {
      "webkit-filter": 'blur(1px)',
      filter: 'blur(1px)'
    }
  }
})
