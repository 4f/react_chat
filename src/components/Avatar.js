import React from 'react'
import MUIAvatar from 'material-ui/Avatar'
import getColor from 'utils/color-from'
import titleInitials from 'utils/title-initials'

import { avatar as applyPropTypes } from 'prop_types/app'
import styles from 'styles/app'


const Avatar = ({ colorFrom, classes, label, ...rest }) => (
  <MUIAvatar className={classes.avatar} style={{ backgroundColor: getColor(colorFrom) }} {...rest}>
    {titleInitials(label)}
  </MUIAvatar>
)

export default styles( applyPropTypes(Avatar) )
