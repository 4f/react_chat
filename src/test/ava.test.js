/* eslint-env jest */
import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Avatar from 'components/ava'

describe('<Avatar />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Avatar colorFrom="12345" label="smit" />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders correctly', () => {
    const tree = renderer.create(<Avatar colorFrom="12345" label="smit" />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
