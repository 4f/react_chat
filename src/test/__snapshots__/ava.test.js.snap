// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> renders correctly 1`] = `
<div
  className="MuiAvatar-root-16 MuiAvatar-colorDefault-17 Avatar-avatar-15"
  style={
    Object {
      "backgroundColor": "#ffc107",
    }
  }
>
  S
</div>
`;
